bundles/admin/admin-announcements/view/AnnouncementsListHandler.js

```diff
@@ -0,0 +1,156 @@
+                this.updateState({
+                    announcements: data.data
+                });
+            }
+        }.bind(this));
+    }
+
+    saveAnnouncement (data) {
+        this.announcementsHelper.saveAnnouncement(data, function (err) {
+            if (err) {
+                Messaging.error(getMessage('messages.saveFailed'));
+            } else {
+                Messaging.success(getMessage('messages.saveSuccess'));
+                this.updateState({
+                    activeKey: []
+                });
+                this.fetchAdminAnnouncements();
+            }
+        }.bind(this));
+    }
+
+    // Update all the announcements f.ex. when saved. Set active key as empty so all panels get closed.
+    updateAnnouncement (data) {
+        this.announcementsHelper.updateAnnouncement(data, function (err, data) {
+            if (err) {
+                Messaging.error(getMessage('messages.updateFailed'));
+                return false;
+            } else {
+                Messaging.success(getMessage('messages.updateSuccess'));
+                removeFromLocalStorageArray(ANNOUNCEMENTS_LOCALSTORAGE, data.id);
+                this.updateState({
+                    activeKey: []
+                });
+                this.fetchAdminAnnouncements();
+            }
+        }.bind(this));
+    }
+
+    deleteAnnouncement (index, id) {
+        this.id = { id };
+        this.announcementsHelper.deleteAnnouncement(this.id, function (err) {
+            if (err) {
+                Messaging.error(getMessage('messages.deleteFailed'));
+            } else {
+                Messaging.success(getMessage('messages.deleteSuccess'));
+                const newList = [...this.state.announcements];
+                newList.splice(index, 1);
+                // Update accordion with announcements and keep all closed
+                this.updateState({
+                    announcements: newList,
+                    activeKey: [],
+                    updated: false
+                });
+                this.fetchAdminAnnouncements();
+            }
+        }.bind(this));
+    }
+    // Cancel creating a new announcement or editing one. Close all panels.
+    cancel (index, id) {
+        if (id !== undefined) {
+            this.updateState({
+                activeKey: []
+            });
+        } else {
+            const newList = [...this.state.announcements];
+            newList.splice(index, 1);
+            this.updateState({
+                announcements: newList,
+                activeKey: []
+            });
+        }
+    }
+
+    addForm () {
+        const newList = [...this.state.announcements];
+        newList.push({
+            title: this.newTitle
+        });
+        this.updateState({
+            announcements: newList,
+            title: this.newTitle
+        });
+    }
+
+    updateTitle (value) {
+        this.updateState({
+            title: value
+        });
+    }
+    toggleActive () {
```

Comment:
Is this similar to updateTitle? The announcement has an active flag that is inherited from the last announcement added to the next one? Consider having a tmpAnnouncementForAdding object in state that triggers the showing of the form and holds the current values. Then it would be obvious what these means and could be reseted between adding announcements by just removing the key from state.

Reply:
Same as title again you might be able to remove this and just use the antd form onFinish()?